packages/block-library/src/navigation/edit/unsaved-inner-blocks.js

```diff
@@ -52,22 +56,29 @@ export default function UnsavedInnerBlocks( {
 		isSaving,
 		draftNavigationMenus,
 		hasResolvedDraftNavigationMenus,
-	} = useSelect( ( select ) => {
-		const {
-			getEntityRecords,
-			hasFinishedResolution,
-			isSavingEntityRecord,
-		} = select( coreStore );
-
-		return {
-			isSaving: isSavingEntityRecord( 'postType', 'wp_navigation' ),
-			draftNavigationMenus: getEntityRecords( ...DRAFT_MENU_PARAMS ),
-			hasResolvedDraftNavigationMenus: hasFinishedResolution(
-				'getEntityRecords',
-				DRAFT_MENU_PARAMS
-			),
-		};
-	}, [] );
+	} = useSelect(
+		( select ) => {
+			if ( isDisabled ) {
+				return {};
```

Comment:
It will return a new object every time and force re-render.

Reply:
Should be fixed 👍